pyodide_build/buildpkg.py

```diff
@@ -60,6 +61,12 @@ def download_and_extract(buildpath, packagedir, pkg, args):
         if not srcpath.is_dir():
             shutil.unpack_archive(str(tarballpath), str(buildpath))
 
+        for extension in ['.tar.gz', '.tgz', '.tar', '.tar.bz2', '.tbz2', '.tar.xz', '.txz', '.zip']:
+            if tarballname.endswith(extension):
+                tarballname = tarballname[:-len(extension)]
+                break
```

Comment:
I think that's equivalent to 

```suggestion
        # strip extension e.g. .tar or .tar.gz
        extracted_dirname = tarballname.with_suffix('').with_suffix('')
```

Reply:
I think `with_suffix` does not work, since `Path('test-v1.0.0.tar').with_suffix('').with_suffix('') == 'test-v1.0'`